rest-service/manager_rest/test/endpoints/test_app.py

```diff
@@ -29,3 +32,75 @@ def test_get_root_404(self):
         """
         resp = self.app.get('/')
         self.assertEqual(404, resp.status_code)
+
+    def test_does_not_init_tracer(self):
+        # Dummy call to invoke the tracer initialization if enabled.
+        self.app.get('/')
+        self.jaeger_mock_config.assert_not_called()
+
+
+@attr(client_min_version=1, client_max_version=base_test.LATEST_API_VERSION)
+class AppWithTracingNoIPTestCase(base_test.BaseServerTestCase):
+    """Test the basic setup of the tracer and dispatch wrapping for the
+    tracing when tracing is enabled but no endpoint IP is set.
+    """
+
+    def setUp(self):
+        super(AppWithTracingNoIPTestCase, self).setUp()
+        # Dummy call to invoke the tracer initialization if enabled.
```

Comment:
Why is this needed?

Reply:
Tracer is initialized before the first request comes in and only then. Therefore I need a request to invoke that.
Why can't I initialize it normally? Because if I do so in the `__init__` or I presume, in the thread that runs the `__init__` of the Flask app, then it hangs. I can explain my theory further if needed.